### PR TITLE
expose stream API in python kernel call interfaces

### PR DESCRIPTION
**Pull request by K-Wu**
Hi there,

I exposed the stream as an option in the python kernel call interfaces. This is needed for my research project at U of I. Please let me know if this could be merged into your main branch with shepherding. 

Thank you.

Best Regards,
Kun

### CODE REVIEW

**Comment by jackkosaian** [comment]
Thanks for your contribution, @K-Wu. I'll work on testing this out and will get back to you with any feedback.

File: python/cutlass/op/conv.py [hunk 1]
```diff
@@ -733,7 +734,8 @@ def run(self, A=None, B=None, C=None, D=None,
             stride=(1, 1), padding=(0, 0), dilation=(1, 1),
             alpha=None, beta=None,
             split_k=("serial", 1), sync: bool = True,
-            print_module: bool = False) -> Conv2dArguments:
+            print_module: bool = False,
+            stream: cuda.CUstream = cuda.CUstream(0)) -> Conv2dArguments:
```
Comment (jackkosaian) [inline python/cutlass/op/conv.py:738]: Could you please update the docstrings for this and other methods that now take in `stream` as an argument?

Comment (K-Wu) [reply python/cutlass/op/conv.py:738]: Hi Jack, sure! This is addressed in commit 91becf9022ce9235d8af6fe7c4e648e6f7e8cb96.

**Comment by K-Wu** [comment]
Hi @jackkosaian, thank you for your review! I have addressed your comment in commit 91becf9022ce9235d8af6fe7c4e648e6f7e8cb96.

Let me know if you need any further changes/clarifications. Thank you again.

**Comment by jackkosaian** [comment]
Could you also add the following inside the `GemmGroupedArguments.__init__` in `python/cutlass/backend/gemm_operation.py`?

```python
        self.stream = kwargs.get("stream", cuda.CUstream(0))
```

This is needed because `GemmGroupedArguments` do not derive from `ArgumentsBase`, and, thus, do not have the `stream` member set.

**Comment by K-Wu** [comment]
Nice catch! I have addressed it in the new commit.

> Could you also add the following inside the `GemmGroupedArguments.__init__` in `python/cutlass/backend/gemm_operation.py`?
> 
> ```python
>         self.stream = kwargs.get("stream", cuda.CUstream(0))
> ```
> 
> This is needed because `GemmGroupedArguments` do not derive from `ArgumentsBase`, and, thus, do not have the `stream` member set.

**Review by hwu36 — APPROVED** [review]

**Pull request merged**